File: src/api/ShowNTell.Domain/Models/Follow.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text;

namespace ShowNTell.Domain.Models
{
    /// <summary>
    /// A model for a user following another user.
    /// </summary>
    public class Follow
    {
        public string UserEmail { get; set; }
        public string FollowerEmail { get; set; }

        public User User { get; set; }
        public User Follower { get; set; }
    }
}

File: src/show-n-tell-vue/src/models/follow.js
function Follow(userEmail, userUsername, followerEmail, followerUsername) {
    this.userEmail = userEmail
    this.userUsername = userUsername
    this.followerEmail = followerEmail
    this.followerUsername = followerUsername
}

Follow.fromJSON = function(follow) {
    return new Follow(
        follow.userEmail,
        follow.userUsername,
        follow.followerEmail,
        follow.followerUsername
    )
}

export default Follow